frontend/src/lib/WebsocketConnection.tsx

```diff
@@ -523,40 +517,53 @@ function doHealthPing(
     )
   }
 
-  xhr.onreadystatechange = () => {
-    if (xhr.readyState !== /* DONE */ 4) {
-      return
-    }
-
-    if (xhr.responseText === "ok") {
-      resolver.resolve(uriNumber)
-    } else if (xhr.status === /* NO RESPONSE */ 0) {
-      retryWhenTheresNoResponse()
-    } else if (xhr.status === 403) {
-      retryWhenIsForbidden()
-    } else {
-      retry(
-        `Connection failed with status ${xhr.status}, ` +
-          `and response "${xhr.responseText}".`
-      )
-    }
-  }
-
-  xhr.ontimeout = e => {
-    retry("Connection timed out.")
-  }
-
   connect = () => {
     const uri = uriList[uriNumber]
     logMessage(LOG, `Attempting to connect to ${uri}.`)
     tryTimestamp = Date.now()
-    xhr.open("GET", uri, true)
 
     if (uriNumber === 0) {
       totalTries++
     }
 
-    xhr.send(null)
+    axios
+      .get(uri, {
+        timeout: timeoutMs,
+      })
+      .then(() => {
```

Comment:
should this say `.then((response) => {` ?

Reply:
IMO, if we're not using the parameter, there's no reason to accept it. (It's not an error to receive fewer params than you're called with.)